include/bluetooth/conn.h

```diff
@@ -209,6 +209,50 @@ int bt_conn_disconnect(struct bt_conn *conn, u8_t reason);
 struct bt_conn *bt_conn_create_le(const bt_addr_le_t *peer,
 				  const struct bt_le_conn_param *param);
 
+#if defined(CONFIG_BT_WHITELIST)
```

Comment:
Please avoid #ifdefs like this in header files. It's perfectly fine to have the declarations here even if the function doesn't get linked in. If the app tries to use it there will be a linker error. The #ifdefs are problematic e.g. because then the doxygen documentation doesn't get pulled in (unless we add extra overrides to the doxygen build).

Reply:
> Please avoid #ifdefs like this in header files. It's perfectly fine to have the declarations here even if the function doesn't get linked in. If the app tries to use it there will be a linker error. The #ifdefs are problematic e.g. because then the doxygen documentation doesn't get pulled in (unless we add extra overrides to the doxygen build).

OK